dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -117,7 +128,21 @@ private void parseRecursive(Reader source, Listener listener) throws IOException
                 break;
             }
             buffer.add((char) readCharacter);
-            if (getStringRepresentation(buffer).endsWith("----")) {
+
+            List<DokuWikiPlugin> componentList = null;
+            if (componentManagerProvider != null) {
+                try {
+                    componentList = componentManagerProvider.get().getInstanceList(DokuWikiPlugin.class);
+                } catch (ComponentLookupException e) {
+                    this.logger.error("Failed to get components", e);
+                }
+            }
+
+            for (DokuWikiPlugin plugin : componentList) {
```

Comment:
`componentList` might be null according to the previous code so you should have a if here (or throw an exception instead of logging an error in the previous code).